src/vs/workbench/contrib/search/browser/searchActionsTextQuickAccess.ts

```diff
@@ -29,6 +34,59 @@ registerAction2(class TextSearchQuickAccessAction extends Action2 {
 
 	override async run(accessor: ServicesAccessor, match: RenderableMatch | undefined): Promise<any> {
 		const quickInputService = accessor.get(IQuickInputService);
-		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX);
+		const searchText = getSearchText(accessor);
+		const searchContents = searchText ? ' ' + searchText : '';
+		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX + searchContents);
 	}
 });
+
+function getSearchText(accessor: ServicesAccessor): string | null {
+	const editorService = accessor.get(IEditorService);
+	const configurationService = accessor.get(IConfigurationService);
+
+	if (editorService.activeTextEditorControl === undefined) {
+		return null;
+	}
+	const activeEditor: IEditor = editorService.activeTextEditorControl;
+	if (!activeEditor) {
```

Comment:
It would be nice to have this conditional and the one below to be combined to reduce code.

Reply:
I don't think this one is resolved yet?